### PR TITLE
Expose ThreadLocalContextStorage publicly

### PR DESCRIPTION
**Pull request by HaloFour**
Exposes the `ThreadLocalContextStorage` enum publicly to enable writing cimple `ContextStorage` implementations without having to rewrite the storage.  Updates the Javadocs for `ContextStorage` to replace the non-existent `Context#threadLocalStorage` method with `ThreadLocalContextStorage.INSTANCE`.

### CODE REVIEW

File: context/src/main/java/io/opentelemetry/context/ContextStorage.java [hunk 1]
```diff
@@ -69,6 +69,15 @@ static ContextStorage get() {
     return LazyStorage.get();
   }
 
+  /**
+   * Returns the default {@link ContextStorage} used to attach {@link Context}s to scopes of
+   * execution. Should only be used when defining your own {@link ContextStorage} in case you want
+   * to delegate functionality to the default implementation.
+   */
+  static ContextStorage threadLocalStorage() {
```
Comment (bogdandrutu) [inline context/src/main/java/io/opentelemetry/context/ContextStorage.java:77]: This makes the method public. What is the reason to do it?

Comment (HaloFour) [reply context/src/main/java/io/opentelemetry/context/ContextStorage.java:77]: To make it easier to write a delegating `ContextStorage` implementation that uses thread-local storage.  This is already suggested by the Javadocs on this type which reference the method `Context#threadLocalStorage` which does not exist.

File: context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java [hunk 2]
```diff
@@ -45,7 +41,8 @@ public Scope attach(Context toAttach) {
 
   @Override
   public Context current() {
-    return THREAD_LOCAL_STORAGE.get();
+    Context threadContext = THREAD_LOCAL_STORAGE.get();
+    return threadContext != null ? threadContext : Context.root();
```
Comment (bogdandrutu) [inline context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java:45]: Changes in this file seem independent of the main issue, consider to split them.

File: context/src/main/java/io/opentelemetry/context/ContextStorage.java (hunk 1 above)
Comment (bogdandrutu) [reply context/src/main/java/io/opentelemetry/context/ContextStorage.java:77]: What about then we make the implementation thread local public enum? I feel that exposing this in the Context ties too much the threadlocal storage to the context.

File: context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java (hunk 2 above)
Comment (HaloFour) [reply context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java]: Done and done, #1974

File: context/src/main/java/io/opentelemetry/context/ContextStorage.java (hunk 1 above)
Comment (HaloFour) [reply context/src/main/java/io/opentelemetry/context/ContextStorage.java:77]: That'd be fine by me.  I was going with the existing convention suggested in the Javadocs on `ContextStorage` as well as by `LazyStorage` (which references a similar helper method on `DefaultContext`).  Should we get others to weigh in before I change this PR?

Comment (bogdandrutu) [reply context/src/main/java/io/opentelemetry/context/ContextStorage.java:77]: Yes, I would like to hear others opinions @anuraaga @jkwatson @carlosalberto

**Comment by HaloFour** [comment]
I've changed the PR so that instead of exposing the `ThreadLocalContextStorage` via a static helper method on `ContextStorage` that the `ThreadLocalContextStorage` enum is exposed publicly and can be used directly.

**Comment by anuraaga** [comment]
I think this PR already happened as a part of https://github.com/open-telemetry/opentelemetry-java/pull/1938 @HaloFour what do you think?

**Comment by HaloFour** [comment]
Yes, that PR does address the same concern.  Thanks!

**Pull request closed without merging**